skimage/restoration/tests/test_j_invariant.py

```diff
@@ -27,10 +28,12 @@ def test_invariant_denoise():
     assert_(denoised_mse < original_mse)
 
 
-def test_invariant_denoise_color():
+@testing.parametrize('dtype', [np.float32, np.float64])
```

Comment:
same thought as previously: why not test all applicable types?

Reply:
Done. There was actually a missing promotion to float32 for float16 inputs!